.circleci/config.yml

```diff
@@ -0,0 +1,135 @@
+version: 2.1
+
+executors:
+  amd64:
+    docker:
+      - image: cimg/base:current
+    resource_class: large
+    environment:
+      PLATFORM: "linux/amd64"
+      EXECUTOR: "amd64"
+  arm64:
+    docker:
+      - image: cimg/base:current
+    resource_class: arm.large
+    environment:
+      PLATFORM: "linux/arm64"
+      EXECUTOR: "arm64"
+
+commands:
+  configure-ghcr:
+    description: Configure GHCR access
+    steps:
+      - run:
+          name: Configure GHCR
+          command: |
+            [ -z "$GHCR_TOKEN" ] && exit 0
+            [ -z "$GHCR_USER" ] && exit 0
+            for i in $(seq 1 5); do
+              echo "Logging into GHCR, attempt $i"
+              echo $GHCR_TOKEN | docker login ghcr.io -u $GHCR_USER --password-stdin && break
+              sleep 1
+            done
+
+  tag-generator:
+    description: Format CIRCLE_TAG
+    steps:
+      - run:
+          name: Generate environment variables for build
+          command: |
+            if [ -n "${CIRCLE_TAG}" ]; then
+              SEMVER=$( echo ${CIRCLE_TAG} | sed -nre 's/^v[^0-9]*(([0-9]+\.)*[0-9]+(-[a-z]+)?).*/\1/p')
+              if [[ -n $SEMVER ]]; then
+                IMAGE_TAG=${SEMVER}
+              else
+                IMAGE_TAG=${CIRCLE_TAG}
+              fi
+            else
+              IMAGE_TAG=latest
+            fi
+            
+            echo -e "export DOCKER_REGISTRY=ghcr.io\n" >> $BASH_ENV
+            echo -e "export DOCKER_ORG=darpa-askem\n" >> $BASH_ENV
+            echo -e "export PACKAGE_NAME=beaker-kernel\n" >> $BASH_ENV
+            echo -e "export IMAGE_TAG=${IMAGE_TAG}\n" >> $BASH_ENV
+            source $BASH_ENV
+
+jobs:
+  build-push-light:
+    parameters:
+      platform:
+        type: executor
+    executor: << parameters.platform >>
+    steps:
+      - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
```

Comment:
This can get very expensive for large images. You'll likely want to check out the estimated credit usage before enabling this.